From 86ee779e22e9fc3a6f3add1935d3392e27df1dca Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Wed, 5 Apr 2023 17:29:04 +0100
Subject: [PATCH 27/41] Make --server=/#/<addr> behave the same as
 --server=<addr>

For consistency with --address and older dnsmasq releases.
---
 src/option.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/option.c b/src/option.c
index 588a558..146ca0c 100644
--- a/src/option.c
+++ b/src/option.c
@@ -3058,8 +3058,8 @@ static int one_opt(int option, char *arg, char *errstr, char *gen_err, int comma
 		    else
 		      flags &= ~SERV_FOR_NODOTS;
 		    
-		    /* address=/#/ matches the same as without domain */
-		    if (option == 'A' && cur_domain[0] == '#' && cur_domain[1] == 0)
+		    /* address=/#/ matches the same as without domain, as does server=/#/.... for consistency. */
+		    if (cur_domain[0] == '#' && cur_domain[1] == 0)
 		      cur_domain[0] = 0;
 		  }
 		
-- 
2.41.0

